src/components/user/uc/users.uc.js

```diff
@@ -0,0 +1,67 @@
+const { ObjectId } = require('mongoose').Types;
+const { GeneralError, NotFound } = require('../../../errors');
+const { userRepo, accountRepo, trashbinRepo } = require('../repo/index');
+const restrictToSameSchool = require('../../../utils/restrictToSameSchool');
+const restrictToRole = require('../../../utils/restrictToRole');
+
+const getUserData = async (id, app) => {
+	const data = {};
+
+	const user = await userRepo.getUser(id, app);
+	if (!(user && user._id && !user.deletedAt)) {
+		throw new NotFound(`User ${id} not found`);
+	}
+	data.user = user;
+
+	const account = await accountRepo.getUserAccount(id, app);
+	if (account) {
+		data.account = account;
+	}
+	return data;
+};
+
+const deleteUserData = async (id, app) => {
+	await accountRepo.deleteUserAccount(id, app);
+};
+
+const createUserTrashbin = async (id, data) => {
+	const trashBin = await trashbinRepo.createUserTrashbin(id, data);
+	if (!(trashBin && trashBin._id)) {
+		throw new GeneralError(`Unable to initiate trashBin`);
```

Comment:
In the most cases GeneralError is not what we want. Is the input of the function wrong then we should throw an error that say it. Otherwise if an error is throw in trashbinRepo.createUserTrashbin it go directly to deleteUserUC

Reply:
Try catch around the operation is the better way if we handle an error on this position.